tests/system/feature_store/test_feature_store.py

```diff
@@ -4813,6 +4813,77 @@ def test_merge_different_number_of_entities(self):
         ).to_dataframe()
         assert_frame_equal(expected_all, df, check_dtype=False)
 
+    @staticmethod
+    def _get_sorted_df(df: pd.DataFrame, sort_column: str):
+        result = (
+            df.reindex(sorted(df.columns), axis=1)
+            .sort_values(by=sort_column)
+            .reset_index(drop=True)
+        )
+        return result
+
+    @pytest.mark.parametrize("engine", ["local", "dask"])
+    def test_parquet_filters(self, engine):
+        parquet_path = os.path.relpath(str(self.assets_path / "testdata.parquet"))
+        df = pd.read_parquet(parquet_path)
+        filtered_df = df.query('department == "01e9fe31-76de-45f0-9aed-0f94cc97bca0"')
+        run_uuid = uuid.uuid4()
+        v3io_parquet_source_path = f"v3io:///projects/{self.project_name}/df_parquet_filtered_source_{run_uuid}.parquet"
+        v3io_parquet_target_path = f"v3io:///projects/{self.project_name}/df_parquet_filtered_target_{run_uuid}"
+        df.to_parquet(v3io_parquet_source_path)
+        parquet_source = ParquetSource(
+            "parquet_source",
+            path=v3io_parquet_source_path,
+            additional_filters=[
+                ("department", "=", "01e9fe31-76de-45f0-9aed-0f94cc97bca0")
+            ],
+        )
+        result = parquet_source.to_dataframe()
+        assert_frame_equal(
+            result.sort_values(by="patient_id").reset_index(drop=True),
+            filtered_df.sort_values(by="patient_id").reset_index(drop=True),
+        )
+        feature_set = fstore.FeatureSet(
+            "parquet-filters-fs", entities=[fstore.Entity("patient_id")]
+        )
+
+        target = ParquetTarget(
+            name="department_based_target",
+            path=v3io_parquet_target_path,
+            partitioned=True,
+            partition_cols=["department"],
+        )
+        feature_set.ingest(source=parquet_source, targets=[target])
+        result = target.as_df(additional_filters=("room", "=", 1)).reset_index(
+            drop=False
+        )
+        # We want to include patient_id in the comparison and to sort both the columns and the values.
+        result = self._get_sorted_df(result, "patient_id")
+        expected = self._get_sorted_df(filtered_df.query("room == 1"), "patient_id")
+        #  reset category type to string:
+        result["department"] = result["department"].astype("str")
+        assert_frame_equal(result, expected)
```

Comment:
```suggestion
        assert_frame_equal(result, expected, check_dtype=False)
```

Reply:
The problem here is that the type was changed to category, if I use check_dtype=False I will still get an error.
I can use check_categorical=False, but in this way it will not check the values of this column.

So I think that the current code is the best option:)

Reply:
added also check_categorical=False